File: 300. Longest Increasing Subsequence.py
class Solution(object):
    def lengthOfLIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        if n == 0:
            return 0
        T = [0 for i in xrange(n)]
        T[0] = 1
        res = 1
        for i in xrange(1, n):
            temp = 1
            for j in xrange(i):
                if nums[i] > nums[j]:
                    temp = max(temp, T[j] + 1)
            T[i] = temp
            res = max(T[i], res)
        return res

File: 300. Longest Increasing Subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var lengthOfLIS = function(nums) {
    if (nums.length === 0)
        return 0;
    let T = [];
    for (let i = 0; i < nums.length; i++)
        T.push(1);
    let res = 1;
    for (let i = 1; i < nums.length; i++)
    {
        for (let j = 0; j < i; j++)
        {
            if (nums[i] > nums[j])
                T[i] = Math.max(T[i], T[j] + 1);
        }
        res = Math.max(T[i], res);
    }
    return res;
};
